### PR TITLE
Use custom-block for docstrings

### PR DESCRIPTION
**Pull request by tpoisot**
The current styling for docstrings isn't really in line with the rest of the vitepress theme - this is a work in progress PR to use `class='jldocstring custom-block'` on the div, to make it look more native.

### CODE REVIEW

**Comment by tpoisot** [comment]
What is the secret sauce to get the site to build using github actions? I'm getting a `final_site` folder on `gh-pages`:

Screenshot_20240916_163811

**Comment by asinghvi17** [comment]
Thanks for the PR!  This looks promising.  

https://tpoisot.github.io/DocumenterVitepress.jl/dev/api has some examples of what this looks like now.

From a brief look it seems that font sizes are smaller in the docstring block than in regular text?

**Comment by avik-pal** [comment]
Would it be possible to mimic the Documenter.jl style for the docstrings here? We can leave out the collapsible and such.

The border was added because DocumenterMarkdown used to generate continuous blocks of docstrings for mkdocs (making it hard to distinguish 2 separate docstrings, except the # marker) which was a constant source of complaint from users.

**Comment by lazarusA** [comment]
`collapsible ` yes, NOT this. I was playing around with them and it just adds overhead most of times. And, in general, I do like a smaller font, probably not too small.

**Comment by tpoisot** [comment]
Thanks all -- I think collapsible (open by default) is definitely achievable, I'm working on it

Font size: this is how `custom-block` are styles by vitepress, but if there's a `.jldocstring` class this can be tweaked

**Comment by lazarusA** [comment]
You could add new styles as in here https://github.com/JuliaDataCubes/YAXArrays.jl/blob/master/docs%2Fsrc%2F.vitepress%2Ftheme%2Fstyle.css#L259

**Comment by tpoisot** [comment]
yup - that's what I'm trying to do but without success so far...

**Comment by tpoisot** [comment]
I get it... for some reason it's pulling the `style.css` from *this* repo and not from my fork

**Comment by lazarusA** [comment]
well... it looks great as it is. Except, we have a huge empty space when collapsed.

**Comment by tpoisot** [comment]
I think that's done?

File: docs/make.jl [hunk 1]
```diff
@@ -32,7 +32,7 @@ makedocs(;
     warnonly = true,
     checkdocs=:all,
     format=DocumenterVitepress.MarkdownVitepress(
-        repo = "github.com/LuxDL/DocumenterVitepress.jl", # this must be the full URL!
+        repo = "github.com/tpoisot/DocumenterVitepress.jl", # this must be the full URL!
```
Comment (lazarusA) [inline docs/make.jl:35]:
```suggestion
        repo = "github.com/LuxDL/DocumenterVitepress.jl", # this must be the full URL!
```
you need to change this one back, before we merge or run the PR.

File: docs/make.jl [hunk 2]
```diff
@@ -61,7 +61,7 @@ makedocs(;
 
 
 deploydocs(; 
-    repo = "github.com/LuxDL/DocumenterVitepress.jl", # this must be the full URL!
+    repo = "github.com/tpoisot/DocumenterVitepress.jl", # this must be the full URL!
```
Comment (lazarusA) [inline docs/make.jl:64]:
```suggestion
        repo = "github.com/LuxDL/DocumenterVitepress.jl", # this must be the full URL!
```
also here.

File: docs/make.jl (hunk 1 above)
Comment (tpoisot) [reply docs/make.jl]: ✅

File: docs/make.jl (hunk 2 above)
Comment (tpoisot) [reply docs/make.jl]: ✅

**Comment by avik-pal** [comment]
Any particular reason to revert the badge (https://github.com/LuxDL/DocumenterVitepress.jl/pull/175/commits/9b637a449864668a4a4b7342eb7b1d01dab3a158)? Personally it looked nicer compared to the dash

**Comment by lazarusA** [comment]
Oh, yeah, I merged thinking the bubble badges were there. We should bring them back.

**Comment by tpoisot** [comment]
They messed with the spacing a little, but I can try to adjust the margins and bring them back!

**Pull request merged**